plugins/block-sharable-procedures/src/observable_parameter_model.ts

```diff
@@ -0,0 +1,129 @@
+/**
+ * @license
+ * Copyright 2022 Google LLC
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+import * as Blockly from 'blockly/core';
+import {triggerProceduresUpdate} from './update_procedures';
+
+/** Represents a procedure parameter. */
+export class ObservableParameterModel implements
+    Blockly.procedures.IParameterModel {
+  private id: string;
+  private variable: Blockly.VariableModel;
+  private shouldFireEvents = false;
+  private procedureModel: Blockly.procedures.IProcedureModel|null = null;
+
+  /**
+   * Constructor for the procedure parameter.
+   * @param workspace The workspace this parameter model exists in.
+   * @param name The name of this parameter.
+   * @param id The optional unique language-neutral ID of the parameter.
+   * @param varId The optional ID of the variable this parameter should be
+   *     associated with.
+   */
+  constructor(
+      private readonly workspace: Blockly.Workspace, name: string, id?: string,
+      varId?: string) {
+    this.id = id ?? Blockly.utils.idGenerator.genUid();
+    this.variable = this.workspace.getVariable(name) ??
+        workspace.createVariable(name, '', varId);
+  }
+
+  /**
+   * Sets the name of this parameter to the given name.
+   * @param name The string to set the name to.
+   * @return This parameter model.
+   */
+  setName(name: string): this {
+    if (name === this.variable.name) return this;
+    const oldName = this.variable.name;
+    this.variable =
+        this.workspace.getVariable(name) ?? this.workspace.createVariable(name);
+    triggerProceduresUpdate(this.workspace);
+    if (this.shouldFireEvents) {
+      // Blockly.Events.fire(
+      //     new (Blockly.Events.get(
+      //         Blockly.Events.PROCEDURE_PARAMETER_RENAME))(
+      //         this.workspace, this.procedureModel, this, oldName));
+    }
+    return this;
+  }
+
+  /**
+   * Unimplemented. The built-in ParameterModel does not support typing.
+   * If you want your procedure blocks to have typed parameters, you need to
+   * implement your own ParameterModel.
+   * @param types The types to set this parameter to.
+   * @throws Throws for the ObservableParameterModel specifically because this
+   *     method is unimplemented.
+   */
+  setTypes(types: string[]): this {
+    throw new Error(
+        'The built-in ParameterModel does not support typing. You need to ' +
+        'implement your own custom ParameterModel.');
+  }
+
+  /**
+   * Returns the name of this parameter.
```

Comment:
This bothered me so I updated the eslint config to allow not having redundant descriptions: https://github.com/google/blockly-samples/pull/1537

if you want to rebase this PR on top of mine, you can remove these. but you'll also have to update your `@return` to `@returns`
otherwise I can update your return statements for you if you submit this before my PR is submitted.

Reply:
It looks like #1537 is good to go in? So I'll just rebase on top of that and update!